src/v/cluster/tests/rm_stm_tests.cc

```diff
@@ -377,7 +377,7 @@ FIXTURE_TEST(test_stale_begin_tx_fenced, rm_stm_test_fixture) {
     };
 
     auto commit_tx = [&](model::tx_seq seq) {
```

Comment:
WDYT about disallowing implicit captures?

Reply:
its definitely convenient in some cases. there are also many cases where lambda is used but something equivalent could have been written inline, so what's the point of explicit captures when either was acceptable. i think for the most part we use explicit captures in core, but tests are more relaxed on requirements?

Reply:
agreed, either way fixed this while I'm here.